### ISSUE TITLE
[BCES01175]Uncharted 3 - dynamic lighting and shadows sometimes does not work

### ISSUE THREAD

**Issue by davewuk** [RPCS3/rpcs3#14164]
### Quick summary

At the start of the chapter "The Middle Way" most of the lighting in the scene comes from a hand held torch. Sometimes the torch completely fails to project any light onto the walls and shadows are not rendered.

### Details

While testing PR #14149 against Master I noticed that the lighting in the scene at the start of the chapter "The Middle Way" isn't rendered correctly. This happens in both the PR and current Master 0.0.28-15330. What is really strange is that the issue mostly occurs when logging is disabled - in fact, I have not managed to capture a log file with the lighting broken as it just doesn't seem to happen when logging is enabled. I've tested both before and after wiping the shader cache and various settings.

When the dynamic lighting is working it looks like this:

![BCES01175_screenshot_2023_07_09_13_12_30](https://github.com/RPCS3/rpcs3/assets/55807111/c2a59e5c-ca64-4ca5-86a1-ac0ea5ba7e3e)

But when it is not working the torch casts no light like this:

![BCES01175_screenshot_2023_07_09_13_18_54](https://github.com/RPCS3/rpcs3/assets/55807111/a611cff7-5c8c-4960-b6d8-b79d5a29fff3)

### Attach a log file

[Working.zip](https://github.com/RPCS3/rpcs3/files/11995161/Working.zip)

I have not been able to capture a log file when the lighting is broken

### Attach capture files for visual issues

Lighting working:

https://1drv.ms/u/s!Ah8-tXJT4RvZgrpQAbl6lUKHXbHT_g?e=Qny5Yv

Lighting not working:

https://1drv.ms/u/s!Ah8-tXJT4RvZgrpRlV8quIz5ccqaqQ?e=657UEh

### System configuration

i7 12700k
16gb DDR4 ram at 3200
RTX 2060 Super 8gb

**Comment by kd-11** [comment]
In the GPU settings tab, set the shader compiler mode to legacy. That should fix any issues caused by the async compiler.

**Comment by davewuk** [comment]
> In the GPU settings tab, set the shader compiler mode to legacy. That should fix any issues caused by the async compiler.

Unfortunately this doesn't seem to fix the issue, I can still reproduce it with logging off and shader compiler set to legacy. I still haven't been able to reproduce the issue with logging on but I'll keep trying.

**Comment by kd-11** [comment]
Also check if it can be reproduced stock. Patches may trigger buggy codepaths in the gamecode as they often re-enable things that the developers did not intend to run.

**Comment by davewuk** [comment]
> Also check if it can be reproduced stock

The game is unplayable without patches I think

![BCES01175_screenshot_2023_07_09_17_51_45](https://github.com/RPCS3/rpcs3/assets/55807111/523406f9-8922-4562-a477-58e76de6aba9)

I will need to test again once the PR is merged to be sure though.

**Comment by davewuk** [comment]
OK that was quicker than I thought. In the latest master build 0.0.28-15334 it is still possible to reproduce the issue with or without patches enabled. However with patches disabled the game is unplayable due to the white blob issue.

Logging enabled, no patches
![BCES01175_screenshot_2023_07_09_17_59_15](https://github.com/RPCS3/rpcs3/assets/55807111/06149cfc-a11f-4546-b404-3641aef76f41)

Logging disabled, no patches
![BCES01175_screenshot_2023_07_09_18_01_46](https://github.com/RPCS3/rpcs3/assets/55807111/975b5131-4846-4210-9a7c-565f050bd933)

Is there anything else I should test? do you want more logs/RSX captures?

**Comment by kd-11** [comment]
No, this is enough to get started I think.

**Comment by Darkhost1999** [comment]
https://github.com/RPCS3/rpcs3/issues/11559
This is 1 of the without patches threads being referenced here.
But they're saying the dynamic lighting and stuff where the other issue makes note of the white splotches on screen.

**Comment by davewuk** [comment]
Well that thread is a complete confusion but ultimately it seems to corroborate what I just said - the white blob issue (which is a complete separate issue to the problems with the dynamic lighting raised here) is **fixed** when you use the patches to play UC3.

**Comment by davewuk** [comment]
Pleased to say that this behaviour no long reproducible in the latest build.
